onboarding/source/amb_light_service.c

```diff
@@ -18,22 +26,91 @@ static void lightServiceTask(void * pvParameters);
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    if (lightServiceTaskHandle == NULL) {
+        // Create controller task
+        xReturned = xTaskCreate(lightServiceTask,           /* Function that implements the task. */
+                                LIGHT_SERVICE_NAME,         /* Text name for the task. */
+                                LIGHT_SERVICE_STACK_SIZE,   /* Stack size in words, not bytes. */
+                                NULL,                       /* Parameter passed into the task. */
+                                LIGHT_SERVICE_PRIORITY,     /* Priority at which the task is created. */
+                                &lightServiceTaskHandle);   /* Used to pass out the created task's handle. */
+    }
+
+    if(xReturned == pdFAIL) return 0;
 
+    if (lightServiceQueueHandle == NULL) {
+        lightServiceQueueHandle = xQueueCreate(LIGHT_SERVICE_QUEUE_SIZE,
+                                                LIGHT_SERVICE_ITEM_SIZE);
+    }
     /* USER CODE END */
     return 1;
 }
 
 static void lightServiceTask(void * pvParameters) {
     /* USER CODE BEGIN */
     // Wait for MEASURE_LIGHT event in the queue and then print the ambient light value to the serial port.
+    ASSERT(lightServiceTaskHandle != NULL);
+    ASSERT(lightServiceQueueHandle != NULL);
+    
+    light_event_t event;
+    BaseType_t xReturned = pdFAIL;
+
+    adcData_t lightData;
+    adcData_t *lightDataPtr = &lightData;
+    while(1) {
+        xReturned = xQueueReceive(lightServiceQueueHandle,
+                                &event,
+                                0);
+        if (xReturned == pdPASS && event == MEASURE_LIGHT) {
+            adcStartConversion(adcREG1, adcGROUP1);
```

Comment:
No need to call both `adcStartConversion` and `adcStartConversion_selChn`. It looks like you're following the example code by TI, so I'd just stick with `adcStartConversion_selChn`.

Reply:
Done